utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -376,13 +377,36 @@ impl<'a, 'b> SysCallsInvocator<'a, 'b> {
             let upper_limit = mem_size.saturating_sub(100);
             let offset = self.unstructured.int_in_range(0..=upper_limit)?;
 
-            vec![
-                // call `gsys::gr_source` with a memory offset
+            let mut ret = {
+                // 3 instructions for invoking `gsys::gr_source` and possibly 3 more
+                // for defining value param so HashWithValue will be constructed.
+                let capacity = 3 + invocable
+                    .has_destination_param_with_value()
+                    .then_some(3)
+                    .unwrap_or_default();
+                Vec::with_capacity(capacity)
```

Comment:
Is it crucial to be that precise here? I.e. does it matter much whether there are not less than 3 or 6 items? If not, then I think this code can be re-written in a more readable way when you put the first 3 items all the time and then if append yet another 3 based on the condition (now you compute that condition twice)

Reply:
done